src/sys/termios.rs

```diff
@@ -1,648 +1,782 @@
+              target_os = "macos",
+              target_os = "netbsd",
+              target_os = "openbsd"))]
+    VDSUSP = libc::VDSUSP as isize,
+    VEOF = libc::VEOF as isize,
+    VEOL = libc::VEOL as isize,
+    VEOL2 = libc::VEOL2 as isize,
+    VERASE = libc::VERASE as isize,
+    #[cfg(any(target_os = "dragonfly", target_os = "freebsd"))]
+    VERASE2 = libc::VERASE2 as isize,
+    VINTR = libc::VINTR as isize,
+    VKILL = libc::VKILL as isize,
+    VLNEXT = libc::VLNEXT as isize,
+    VMIN = libc::VMIN as isize,
+    VQUIT = libc::VQUIT as isize,
+    VREPRINT = libc::VREPRINT as isize,
+    VSTART = libc::VSTART as isize,
+    #[cfg(any(target_os = "dragonfly",
+              target_os = "freebsd",
+              target_os = "macos",
+              target_os = "netbsd",
+              target_os = "openbsd"))]
+    VSTATUS = libc::VSTATUS as isize,
+    VSTOP = libc::VSTOP as isize,
+    VSUSP = libc::VSUSP as isize,
+    #[cfg(target_os = "linux")]
+    VSWTC = libc::VSWTC as isize,
+    #[cfg(target_os = "haiku")]
+    VSWTCH = libc::VSWTCH as isize,
+    VTIME = libc::VTIME as isize,
+    VWERASE = libc::VWERASE as isize,
+    #[cfg(target_os = "dragonfly")]
+    VCHECKPT = libc::VCHECKPT as isize,
+}
 
-        bitflags! {
-            pub struct OutputFlags: tcflag_t {
-                const OPOST  = 0x00000001;
-                const ONLCR  = 0x00000004;
-            }
-        }
+pub use libc::NCCS;
+#[cfg(any(target_os = "dragonfly",
+          target_os = "freebsd",
+          target_os = "linux",
+          target_os = "macos",
+          target_os = "netbsd",
+          target_os = "openbsd"))]
+pub use libc::_POSIX_VDISABLE;
+
+libc_bitflags! {
+    /// Flags for configuring the input mode of a terminal
+    pub flags InputFlags: tcflag_t {
+        IGNBRK,
+        BRKINT,
+        IGNPAR,
+        PARMRK,
+        INPCK,
+        ISTRIP,
+        INLCR,
+        IGNCR,
+        ICRNL,
+        IXON,
+        IXOFF,
+        IXANY,
+        IMAXBEL,
+        #[cfg(any(target_os = "android", target_os = "linux", target_os = "macos"))]
+        IUTF8,
+    }
+}
 
-        bitflags! {
-            pub struct ControlFlags: tcflag_t {
-                const CSIZE      = 0x00000030;
-                const CS5        = 0x00000000;
-                const CS6        = 0x00000010;
-                const CS7        = 0x00000020;
-                const CS8        = 0x00000030;
-                const CSTOPB     = 0x00000040;
-                const CREAD      = 0x00000080;
-                const PARENB     = 0x00000100;
-                const PARODD     = 0x00000200;
-                const HUPCL      = 0x00000400;
-                const CLOCAL     = 0x00000800;
-                const CRTSCTS    = 0x80000000;
-                #[cfg(target_os = "android")]
-                const CBAUD      = 0o0010017;
-            }
-        }
+// FIXME: Get these types changed to all `tcflag_t` in `libc` so this can use the `libc_bitflags` macro
```

Comment:
It looks like most of these constants are already defined as `tcflag_t`.  Would it be possible to use the `libc_bitflags` macro now, if you added `as libc::tcflag_t` to the stragglers?